src/twister2/builder/build_manager.py

```diff
@@ -96,6 +105,10 @@ def build(self) -> None:
         """
         Build source code.
         """
+        if self.build_filter_processor:
+            cmake_extra_args: list[str] = self.build_filter_processor.process()
+            self.builder.extend_cmake_extra_args(cmake_extra_args)
```

Comment:
I think that this passing `cmake_extra_args` is unnecessary. Moreover, I think that storing `cmake_extra_args` as an attribute in `BuilderAbstract` class is also unnecessary - both `WestBuilder` and `CmakeBuilder` call `_prepare_cmake_args` method to generate this extra args and as a result `self.cmake_extra_args` are not used in any place.

Reply:
Done